Source/scrollrt.cpp

```diff
@@ -943,7 +943,7 @@ static void Zoom(CelOutputBuffer out)
 			*dst-- = *src;
 			src--;
 		}
-		memcpy(dst - out.pitch(), dst, wdt * 2 + 1);
+		memcpy(dst - out.pitch(), dst, wdt * 2);

```

Comment:
There that fixes it:
```suggestion
		memcpy(dst - out.pitch() + 1, dst + 1, wdt * 2);
```

Reply:
This makes the left side look wrong. I'll look into this a bit more (I think 1 should be added but not always)

Reply:
I got here because I tried removing BORDER/SCREEN_XY paddings and got a `memcpy` overlap ASAN warning when I did that

Reply:
You probably can't get rid of BORDER before all blotting happens via SDL, unless you modefy all biltter functions to be aware of the buffer edges.

Else sprites leaving on the left will appear on the right and vice versa (shifted 1 px in height).

Reply:
> This makes the left side look wrong. I'll look into this a bit more (I think 1 should be added but not always)

Are you sure about that, it looked correct when I tested it.

Reply:
Maybe there is some integer rounding going on that needs to be taking in to consideration and various resolutions (I was testing at 640x480)

Reply:
I've had a detailed look at this function and it had a few issues.
I rewrote it with comments and fixed this issue and an issue with odd dimensions.